### DISCUSSION TITLE
Aborting Merges

### DISCUSSION THREAD

**Discussion by meniku** [gitextensions/gitextensions#9575] [Q&A]
Why does Git extensions use `reset --hard` to abort merges and not just `merge --abort`?
I know there is a warning that I could have read, but I just didn't expect that and it just destroyed a bunch of local modifications.

**Comment by mstv** [comment]
This is a valid question. (I have not noticed such a warning.)

Where in GE did you abort the merge?

The "merge in progress" banner of FormBrowse does use:
```
       public static ArgumentString AbortMergeCmd()
        {
            return new GitArgumentBuilder("merge") { "--abort" };
        }
```

Any objections to switch to `merge --abort` instead of `reset --hard` all over GE, @gitextensions/git-extensions-team?

**Reply by RussKie** [reply]
I think that'd be the right move.

**Reply by jbialobr** [reply]
That would be definitely a good change. I believe the command hadn't been available when it was implemented in GE. I would consider adding the autostash option to the merge dialog - we have it in the pull dialog.

**Reply by mstv** [reply]
The button in the conflict resolution dialog will get clearer texts with #9942.
The merge/rebase can be aborted using the regarding banner button in the Revision Grid after closing the dialog.
